rsfbclient-rust/src/wire.rs

```diff
@@ -241,18 +247,23 @@ pub fn transaction_operation(tr_handle: u32, op: TrOp) -> Bytes {
 }
 
 /// Execute immediate request
-pub fn exec_immediate(tr_handle: u32, dialect: u32, sql: &str) -> Bytes {
+pub fn exec_immediate(
+    tr_handle: u32,
+    dialect: u32,
+    sql: &str,
+    charset: &Charset,
+) -> Result<Bytes, FbError> {
     let mut req = BytesMut::with_capacity(28 + sql.len());
```

Comment:
The capacity calculation is counting always the `UTF-8` byte count, so it may be wrong in other encodings.

Reply:
Fixed!